ui/raidboss/data/03-hw/dungeon/gubal_library_hard.js

```diff
@@ -0,0 +1,300 @@
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      infoText: {
+        en: 'AoE',
+      },
+    },
+    {
+      // Ferrofluid handling is shamelessly copied from Hades Normal
+      id: 'Gubal Hard Ferrofluid Collect',
+      regex: / 1B:\y{ObjectId}:(\y{Name}):....:....:(0030|0031):/,
+      run: function(data, matches) {
+        data.fluid = data.fluid || {};
+        if (matches[2] == '0030')
+          data.fluid[matches[1]] = 'positive';
+        if (matches[2] == '0031')
+          data.fluid[matches[1]] = 'negative';
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid Execute',
+      regex: / 1B:\y{ObjectId}:\y{Name}:....:....:(?:0030|0031):/,
+      suppressSeconds: 5,
+      delaySeconds: 0.5,
+      infoText: function(data) {
+        if (data.fluid[data.me] == data.fluid['Liquid Flame']) {
+          return {
+            en: 'Close to boss',
+          };
+        }
+        return {
+          en: 'Away from boss',
+        };
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid Cleanup',
+      regex: / 1B:\y{ObjectId}:\y{Name}:....:....:003[01]:/,
+      delaySeconds: 5,
+      run: function(data) {
+        delete data.fluid;
+      },
+    },
+    {
+      id: 'Gubal Hard Slosh',
+      regex: / 23:\y{ObjectId}:Liquid Flame:\y{ObjectId}:(\y{Name}):....:....:0039/,
+      condition: function(data, matches) {
+        return data.me == matches[1];
+      },
+      infoText: {
+        en: 'Away from boss',
+      },
+    },
+    {
+      id: 'Gubal Hard Seals',
+      regex: / 1A:\y{ObjectId}:(\y{Name}) gains the effect of (Moon|Sun)seal/,
+      condition: function(data, matches) {
+        return data.me == matches[1];
+      },
+      infoText: function(data, matches) {
+        if (matches[2] == 'Moon') {
+          return {
+            en: 'Stand in blue',
+          };
+        }
+        return {
+          en: 'Stand in red',
+        };
+      },
+    },
+    {
+      // This inflicts a vulnerability stack on the tank if not interrupted
+      id: 'Gubal Hard Condensed Libra',
+      regex: / 14:198D:Mechanoscribe starts using Condensed Libra on \y{Name}/,
+      infoText: function(data) {
+        if (data.CanStun()) {
+          return {
+            en: 'Stun Mechanoscribe',
+          };
+        }
+        if (data.CanSilence()) {
+          return {
+            en: 'Interrupt Mechanoscribe',
+          };
+        }
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Quakes',
+      regex: / 14:1956:Strix starts using On The Properties Of Quakes/,
+      infoText: {
+        en: 'Stand in light circle',
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Tornadoes',
+      regex: / 14:1957:Strix starts using On The Properties Of Tornados/,
+      infoText: {
+        en: 'Stand in dark circle',
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Imps',
+      regex: / 14:1959:Strix starts using On The Properties Of Imps/,
```

Comment:
What do you think about having a delaySeconds on this to account for the difference in time between when the cast starts and when you get the imp effect?

Reply:
The only issue I have with this is that it's *really* easy to end up too far from the cleanse point if the user isn't thinking about it, and if there's a delay the player might end up Imped for more than a GCD while they chase down a cleanse. As it is, if the user is in the center of the room, and they move as soon as the trigger fires, they will arrive at the cleanse point about 1.5-2s before the cast completes. Maybe "Green after Imp" would convey the intent better?

Reply:
No, that's fair.  I guess it's better to warn ahead of time, and so this is fine as it is.